sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/OrderedFilters.scala

```diff
@@ -94,7 +91,7 @@ class CSVFilters(filters: Seq[sources.Filter], requiredSchema: StructType)
     predicate != null && !predicate.eval(row)
```

Comment:
Max, should we maybe add a variable that increase one by one and assert against `index`? We can set it to `0` in `reset`. My biggest worry in here and `JsonFilters` is that it's a bit error-prone ..

Reply:
>  My biggest worry in here and JsonFilters is that it's a bit error-prone ..

Why? Could you explain.

Reply:
> Max, should we maybe add a variable that increase one by one and assert against index?

Yeh, we can add such assert. By the way, both users of the class - CSV and Avro have a loop in which we iterate over schema field indexes from 0 to `index`. The assert will never trigger actually. It could be useful when you are going to support filters pushdown for new datasource, and just accidentally use  the ordered filters but this is mostly a design issue.